client/ayon_resolve/api/pipeline.py

```diff
@@ -97,27 +99,33 @@ def get_workfile_extensions(self):
     def get_containers(self):
         return ls()
 
+    def get_context_data(self):
+        return {}
+
+    def update_context_data(self, data, changes):
+        pass
```

Comment:
This is still lacking - and due to this any context option attributes in the publisher UI will not persist. Which isn't a big problem because apparently we do not have any context attributes exposed yet in the publisher UI, at least I didn't see any in resolve - but we might want to add todos here.

```suggestion
    def get_context_data(self):
        # TODO: implement to support persisting context attributes
        return {}

    def update_context_data(self, data, changes):
        # TODO: implement to support persisting context attributes
        pass
```

Reply:
It'd basically be an implementation similar to this https://github.com/ynput/ayon-resolve/blob/1b6e9aa353dc88d86d51ec3f41920e27ee0e441d/client/ayon_resolve/plugins/create/create_workfile.py#L24-L46 but then storing it in something that does not clash with that data.

Reply:
This creator class is inherited by "exposed" AYON creators implemented in `plugins/create/create_shot_clip.py` file.
I believe the code you are looking for is there:
https://github.com/ynput/ayon-resolve/blob/feature/AY-979_finish_Resolve_new_publisher/client/ayon_resolve/plugins/create/create_shot_clip.py#L120

Reply:
No - the context data is the "global context data" of the Create context and does not store "per instance". This would still need implementation exactly here on the Host addon with the `IPublishHost` interface

See: https://ayon.ynput.io/docs/dev_publishing#required-functions-in-host-implementation

In Maya we store that in the global file info for example, see [here](https://github.com/ynput/ayon-maya/blob/ca40abe992efce70d9ef26f049a5500513ba0ea7/client/ayon_maya/api/pipeline.py#L158-L170).

Reply:
OK, I'll report the TODOs then.